blocklib/picoscope3000a/picoscope3000a/picoscope3000a_cpu.cc

```diff
@@ -754,9 +734,7 @@ std::error_code picoscope_3000a_impl::set_buffers(size_t samples, uint32_t block
             block_number,
             convert_to_ps3000a_ratio_mode(d_downsampling_mode));
         if (status != PICO_OK) {
-            GR_LOG_ERROR(d_logger,
-                         "ps3000aSetDataBuffer (port " + std::to_string(port) +
-                             "): " + ps3000a_get_error_message(status));
+            d_logger->error("ps3000aSetDataBuffer (port {}): {}", port, detail::get_error_message(status));
```

Comment:
@RalphSteinhagen usually prefers the whole PR to be squashed into one commit, so I tend to not worry about the separation, i.e. often commits are squashed already, like here, or I add incremental commits which I would usually squash, as it's only temporary anyway. (The alternative is rewriting the history constantly, which is annoying as soon as someone actually starts testing /reviewing)

Reply:
I also noticed some more porting issues with all the PS blocks, which I tend to fix as part of this PR, to have a clean state with working tests.

Reply:
> The alternative is rewriting the history constantly, which is annoying as soon as someone actually starts testing /reviewing

Or to open a separate PR for it in advance. Though yea, now it is a bit late. Not a big deal … just wanted to mention that it would be nice.